### PR TITLE
feat(amplify-category-geo): add ability to grant permissions for Maps and Search to cognito groups

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

Adds Walkthrough to enable developer to grant permissions for Maps, Search to users of cognito groups.

#### Issue #, if available
https://github.com/aws-amplify/amplify-cli/issues/7850

#### Description of how you validated changes
unit tests and tested in a JS app

#### Screenshots and updated DX
When adding a Map or Search Index, if the Amplify backend already has one or more cognito groups added, we will give the developer choice to grant Read only permissions to cognito groups in addition to the Auth/UnAuth roles. This is necessary since Geofencing capability of geo only supports Group based access aimed to be used by Geofence Admin users and they might also need to access the Map and/or Search capabilities.
The Walkthrough will look like this for `update` and is the same for `add` command:
image

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Amplifiyer** [comment]
@phani-srikar Can you add screenshots of the updated walkthrough and DX

**Comment by phani-srikar** [comment]
> @phani-srikar Can you add screenshots of the updated walkthrough and DX in the PR description
Added

**Comment by Amplifiyer** [comment]
What will be the DX if the maps/search is already added and the developer now adds a geofence with a cognito group.  While adding geofence permissions to a cognito group, should we ask developers if they want to give existing maps/search access to this cognito group?

**Comment by phani-srikar** [comment]
The DX will be to update the added Map/Search Index to give access to that cognito group. Cross resource modifications although might make some cases easier for developers, but could be risky especially when dealing with access permissions. We will make it clear in the documentation for Geofencing about how to add access for Maps/Search to a cognito group.

File: packages/amplify-category-geo/src/provider-controllers/placeIndex.ts [hunk 1]
```diff
@@ -35,21 +35,12 @@ export const updatePlaceIndexResource = async (
...
-      name: placeIndexParams.name,
-      isDefault: placeIndexParams.isDefault,
-      pricingPlan: placeIndexParams.pricingPlan
-    });
-  }
-  else {
-    throw insufficientInfoForUpdateError(ServiceName.PlaceIndex);
-  }
+  const completeParameters: PlaceIndexParameters = convertToCompletePlaceIndexParams(placeIndexParams);
+  modifyPlaceIndexResource(context, completeParameters);
```
Comment (attilah) [inline packages/amplify-category-geo/src/provider-controllers/placeIndex.ts:39]: missing `await`

**Review by attilah — APPROVED** [review]

**Pull request merged**